src/v/security/gssapi.h

```diff
@@ -0,0 +1,191 @@
+/*
+ * Copyright 2023 Redpanda Data, Inc.
+ *
+ * Licensed as a Redpanda Enterprise file under the Redpanda Community
+ * License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://github.com/redpanda-data/redpanda/blob/master/licenses/rcl.md
+ */
+#pragma once
+
+#include "bytes/bytes.h"
+
+#include <gssapi/gssapi.h>
+#include <gssapi/gssapi_ext.h>
+
+#include <string_view>
+
+namespace security::gss {
+
+class buffer_view {
+public:
+    explicit buffer_view(::gss_buffer_desc bd)
+      : _impl{bd} {}
+    explicit buffer_view(std::string_view sv)
+      : _impl{sv.size(), const_cast<char*>(sv.data())} {}
+    explicit buffer_view(bytes_view bv)
+      : _impl{bv.size(), const_cast<unsigned char*>(bv.data())} {}
+
+    explicit operator std::string_view() const {
+        return {static_cast<char*>(_impl.value), _impl.length};
+    };
+    explicit operator bytes_view() const {
+        return {static_cast<unsigned char*>(_impl.value), _impl.length};
+    };
+
+    const void* value() const { return _impl.value; }
+    size_t size() const { return _impl.length; }
+    const gss_buffer_desc* operator&() const { return &_impl; };
+    // TODO BP: Should I wrap every call that only uses this as an input param?
+    gss_buffer_desc* operator&() { return &_impl; };
+
+private:
+    ::gss_buffer_desc _impl;
+};
+
+class buffer : public buffer_view {
+public:
+    using buffer_view::buffer_view;
+    explicit buffer()
+      : buffer_view{::gss_buffer_desc{0, nullptr}} {}
+    buffer(buffer&&) = delete;
+    buffer(const buffer&) = delete;
+    buffer operator=(buffer&&) = delete;
+    buffer operator=(buffer const&) = delete;
+
+    ~buffer() {
+        if (value() != nullptr) {
+            OM_uint32 minor{};
+            gss_release_buffer(&minor, &(*this));
```

Comment:
nit:

this looks so error prone to me like `&(*this)` isn't what it should be. of course, because of the operator overload it is. but it makes it look like i should be able to `gss_release_buffer(&minor, this)`. anyway, wrapping c apis is always fun. maybe do this in the super class destructor and access impl directly?